src/algos/perfect_foresight.jl

```diff
@@ -0,0 +1,117 @@
+
+function perfect_foresight(model, exo; T=200, verbose=true, complementarities=true)
```

Comment:
It looks like we assume `exo` is a matrix of numbers. 

if that is correct, I'd like to put that in the signature as a form of documentation